### PR TITLE
Ability to have parent event id inside response message

### PR DESCRIPTION
**Pull request by Topru333**

### CODE REVIEW

File: src/test/kotlin/com/exactpro/th2/http/client/ResponseTest.kt [hunk 1]
```diff
@@ -0,0 +1,107 @@
+package com.exactpro.th2.http.client
```
Comment (ivandruzhinin) [inline src/test/kotlin/com/exactpro/th2/http/client/ResponseTest.kt:1]: I don't see a need to involve network here. We should rather test that Th2RawHttpRequest behaves correctly e.g. its with* method returns an instance which retains parent event id

**Review by ivandruzhinin — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/exactpro/th2/http/client/util/MessageUtil.kt [hunk 2]
```diff
@@ -167,12 +174,12 @@ private fun HttpMessage.toBatch(connectionId: ConnectionID, direction: Direction
             if (contentTypes.isNotEmpty()) this[CONTENT_TYPE_PROPERTY] = contentTypes
         }
     }
-
+    val parentEventID = if (request is Th2RawHttpRequest) request.parentEventId else null
```
[the remark this answers is not in the document]
Comment (Topru333) [reply src/main/kotlin/com/exactpro/th2/http/client/util/MessageUtil.kt:177]: thx for good example

File: src/main/kotlin/com/exactpro/th2/http/client/api/Th2RawHttpRequest.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    override fun withRequestLine(requestLine: RequestLine): RawHttpRequest {
+        val newHost = checkNotNull(RawHttpHeaders.hostHeaderValueFor(requestLine.uri)) {"RequestLine host must not be null"}
+        val headers: RawHttpHeaders = if (newHost.equals(headers.getFirst("Host").orElse(""), true)) {
+            headers
+        } else {
+            RawHttpHeaders.newBuilderSkippingValidation(headers)
+                .overwrite("Host", newHost)
+                .build()
+        }
+        return Th2RawHttpRequest(
```
[the remark this answers is not in the document]
Comment (Topru333) [reply src/main/kotlin/com/exactpro/th2/http/client/api/Th2RawHttpRequest.kt:39]: val newHost = RawHttpHeaders.hostHeaderValueFor(requestLine.uri) ?: error("RequestLine host must not be null")
  This one gonna throw java.lang.IllegalStateException: () -> kotlin.String coz Kotlin can't process return type of error

[the remark this answers is not in the document]
Comment (Topru333) [reply src/main/kotlin/com/exactpro/th2/http/client/api/Th2RawHttpRequest.kt]: val newHost = RawHttpHeaders.hostHeaderValueFor(requestLine.uri) ?: error("RequestLine host must not be null")
This one gonna throw java.lang.IllegalStateException: () -> kotlin.String (in runtime) coz Kotlin can't process return type of error

**Review by ivandruzhinin — APPROVED** [review]

**Pull request merged**